### PR TITLE
Add .circleci and lint the code

### PR DESCRIPTION
**Pull request by ShipraShalini**
- Uses stable Python and Alpine versions instead of latest.
- Integrates CircleCI
- Sorts imports and formats code.
- Adds basic project setups like `pyproject.toml` and `pre-commit-config.yml`.
- Adds `requirements-dev.txt` for dev env dependencies.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,15 @@
-FROM python:3.10.0a4-alpine3.12
```
Comment (prachipathrayan) [inline Dockerfile:1]: this image pull is not working

File: Dockerfile [hunk 2]
```diff
@@ -1,11 +1,15 @@
-FROM python:3.10.0a4-alpine3.12
-RUN apk update && apk upgrade && \
-    apk add --no-cache bash git
-RUN apk add build-base
+FROM python:3.7.4-alpine3.12
```
Comment (prachipathrayan) [inline Dockerfile:1]: this image is not working

Comment (ShipraShalini) [reply Dockerfile:1]: Handled, please review again.

**Review by vaibhav353 — COMMENTED** [review]
There are a lot of unnecessary diffs in the PRs, maybe your editor is doing it? This makes it difficult to review.
For example - See listMembers.py, logger.py

**Review by vaibhav353 — COMMENTED** [review]
As the code change in th PR touches many files, let's test all the features before merging it to the main.

**Comment by vedansh** [comment]
@vaibhav353 we have checked, everything is working

**Review by prachipathrayan — APPROVED** [review]
Tested all commands with backend, everything is working fine.

**Pull request merged**